internal/services/blackwhitelist_test.go

```diff
@@ -86,151 +84,205 @@ func TestGenerateProjects(t *testing.T) {
 			Project:     "native",
 			Environment: "development",
 		},
+		{
+			Project:     "native",
+			Environment: "integration",
+		},
+		{
+			Project:     "native",
+			Environment: "production",
+		},
 	}
 
+
+
 	//WHITELIST
-	t.Run("with_empty_whitelist", func(t *testing.T) {
+	t.Run("whitelist_enabled_and_projects_in_blacklist", func(t *testing.T) {
 
 		blackWhitelistData := map[string]string{
 			"blacklist": "native-development,native-integration",
 			"whitelist": "",
 		}
 
-		blackWhitelist := services.MakeBlackWhitelist(blackWhitelistData)
+		expected := []*types.Project{
+			{
+				Project:     "native",
+				Environment: "development",
+			},
+			{
+				Project:     "native",
+				Environment: "integration",
+			},
+		} 
+
+		blackWhitelist := MakeBlackWhitelist(blackWhitelistData)
 
 		utils.Config = &types.Config{Whitelist: true}
-		result := GenerateProjects(fakeProject, &blackWhitelist)
-		assert.Equal(t, []bool{false}, result)
+		_,got,_ := GenerateProjects(fakeProject, &blackWhitelist)
+		assert.ElementsMatch(t,got,expected, "the project is in deleted projects ")
 	})
 
-	t.Run("with_equal_whitelist", func(t *testing.T) {
+	t.Run("whitelist_enabled_and_project_is_on_both_lists", func(t *testing.T) {
```

Comment:
nit: i think the name of the test is not obvious. It does not explain the test, it explains the input of the test. You can change that on a follow-up commit.

Example of test name. "blacklist_even_when_on_whitelist"